### PR TITLE
JSON API updates

### PR DESCRIPTION
**Pull request by imtayadeway**
A couple of updates to ensure that:

* users can be updated with the proper JSON API format
* the 'application/vnd.api+json' mimetype is registered so that JSON API requests get properly deserialized and inserted into the `params`

There is probably similar work to be done on the teams controller but I might try to do that in a follow up if that's OK (I'll need to write tests for that and this is the only change that's required by my current work on the user profile stuff)

### CODE REVIEW

**Comment by imtayadeway** [comment]
@lbaillie Added a lot more updates to this! Basically:

* AMS was properly serializing keys in dash-case, but we weren't expecting them to be dash-case when deserializing. I used AMS::Deserialization to handle this
* Some of our tests weren't sending requests in the proper JSONAPI format - fixed
* I've also updated some things to get foreign keys from the URL where appropriate. I'll need to check if this breaks anything in the UI. If it does I think we should fix that as I think it will be better in the long run. Let's discuss if you think this could be a problem
* I think we need more testing here (I think you added some issues to this effect), but hard to tell where the scope of this PR ends. I think it could be good to leave it here (even though I can see more work that needs to be done) if it provides all the support for the user profile feature in the front-end, and I add some follow ups to this when that's done. WDYT?

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -55,9 +55,9 @@ def ensure_correct_user
 
     # Only allow a trusted parameter "white list" through.
     def user_params
-      params
-        .require(:data)
-        .require(:attributes)
-        .permit(:email, :username, :profile_text, :team_id, :provider, :uid)
+      ActiveModelSerializers::Deserialization.jsonapi_parse(
+        params,
+        only: [:email, :username, :"profile-text", :team_id, :provider, :uid]
```
Comment (lbaillie) [inline app/controllers/users_controller.rb]: Should `:team_id` also be dasherized here?

File: app/controllers/teams_controller.rb [hunk 2]
```diff
@@ -47,6 +47,9 @@ def set_team
 
     # Only allow a trusted parameter "white list" through.
     def team_params
-      params.permit(:team_name, :user_id)
+      ActiveModelSerializers::Deserialization.jsonapi_parse(
+        params,
+        only: [:"team-name", :user_id]
```
Comment (lbaillie) [inline app/controllers/teams_controller.rb]: Same with `:user_id` here - should that be dasherized?

**Comment by lbaillie** [comment]
Agreed, I can definitely also see there is more work to be done, and some of the tests could be a bit more robust, but I don't want the scope of this PR to become too out of hand, so I'm going to test the UI manually just to make sure everything works as expected and if it's all good, I will probably merge this as-is and we can work on the other stuff separately 😄

**Comment by lbaillie** [comment]
Update: so... I updated my OS to Sierra this week and now I'm having all kinds of problems bundling. I think I just need to update some dependencies but sadly, airplane wi-fi is just not built for that 😫 but hopefully I'll be able to figure it out this weekend or Monday morning and be able to actually manually check everything out 👍

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (imtayadeway) [inline app/controllers/users_controller.rb]: Technically yes, and I was very tempted to! :grin: But I have a couple of thoughts about this and the one below:

1. If it's being sent in cameCase, it's not any more broken if I leave it like this :wink: 
2. If I do fix it, I want to add a test for it with the right usecase, and I think that will take it this PR out of scope 
3. Also! I have different ideas about this, and want to remove it, but more on that later.

So I think (if that's OK) I'll leave this as is and you will definitely get some follow up PRs to address these one way or another!

**Comment by imtayadeway** [comment]
@lbaillie oh boo! Boo to you, Sierra. Well of course that is no problem! Whenever you can is fine :grin:

**Comment by lbaillie** [comment]
YAY IT WORKS!! I finally got my local dev environment working again and can confirm this works with the `user-show-update` branch on the Ember side, so I'm going to take one last look at that PR and then probably merggggeeee 🎉 🎉 🎉

**Pull request merged**